pkg/api/v20160330/validate.go

```diff
@@ -122,7 +122,7 @@ func validatePoolName(poolName string) error {
 }
 
 func validateDNSName(dnsName string) error {
-	dnsNameRegex := `^([a-z][a-z0-9-]{1,45}[a-z0-9])$`
```

Comment:
the change is, {1,45} -> {1,43}

As the message is saying "The DNS name must contain between 3 and 45 characters", we need to count the first and last ones.